tensorflow/python/ops/summary_ops_v2.py

```diff
@@ -197,14 +197,32 @@ class SummaryWriter(object):
   """Interface representing a stateful summary writer object."""
 
   @abc.abstractmethod
-  def set_as_default(self):
-    """Enables this summary writer for the current thread."""
+  def set_as_default(self, step=None):
+    """Enables this summary writer for the current thread.
+
+    For the `step` argument, see `tf.summary.experimental.set_step()`.
+
+    Args:
+      step: An `int64`-castable default step value, or `None`.
+        When not `None`, the current step is modified to the given value.
+        When `None`, the current step is not modified.
+    """
     raise NotImplementedError()
 
   @abc.abstractmethod
   @tf_contextlib.contextmanager
-  def as_default(self):
-    """Returns a context manager that enables summary writing."""
+  def as_default(self, step=None):
+    """Returns a context manager that enables summary writing.
+
+    For the `step` argument, see `tf.summary.experimental.set_step()`.
+
+    Args:
+      step: An `int64`-castable default step value, or `None`.
+        When not `None`, the current step is captured, replaced by a given one,
+        and the original one is restored when the context manager exists.
```

Comment:
nit: when the context manager **exits**

Reply:
Thanks, fixed.

Reply:
@foxik thank you for the changes, please also add tests for this functionality

Reply:
@jsimsa I will try, but I am not sure about the required scope, as `tf.summary.experimental.{get,set}_step` nor `SummaryWriter`s do not have any tests.

Is it enough to check that after using `set_as_default` and `as_default`, the `get_step` returns what it should? (Plus restoration of the step after `as_default` exists?)

Reply:
Added tests that check the real generated event logs.